cli/src/main/java/com/devonfw/tools/ide/io/FileAccessImpl.java

```diff
@@ -282,22 +284,74 @@ private void copyRecursive(Path source, Path target, FileCopyMode mode) throws I
     }
   }
 
+  @Override
+  public void makeSymlinkRelative(Path link) {
+
+    makeSymlinkRelative(link, false);
+  }
+
+  @Override
+  public void makeSymlinkRelative(Path link, boolean followTarget) {
+
+    if (!Files.isSymbolicLink(link)) {
+      throw new IllegalStateException(
+          "Can't call makeSymlinkRelative on " + link + " since it is not a symbolic link.");
+    }
+    Path linkTarget = null;
+    try {
+      linkTarget = followTarget ? link.toRealPath() : Files.readSymbolicLink(link);
+    } catch (IOException e) {
+      throw new RuntimeException("For link " + link + " the call to "
+          + (followTarget ? "toRealPath" : "readSymbolicLink") + " in method makeSymlinkRelative failed.", e);
+    }
+    this.context.getFileAccess().delete(link); // delete old absolute link
+    this.context.getFileAccess().relativeSymlink(link, linkTarget); // and replace it by the new relative link
+  }
+
+  @Override
+  public void relativeSymlink(Path link, Path source) {
+
+    Path relativeSource = link.getParent().relativize(source);
+    // to make relative links like this work: dir/link -> dir
+    relativeSource = (relativeSource.toString().isEmpty()) ? Paths.get(".") : relativeSource;
+    symlink(relativeSource, link);
+  }
+
   @Override
   public void symlink(Path source, Path targetLink) {
 
     this.context.trace("Creating symbolic link {} pointing to {}", targetLink, source);
     try {
-      if (Files.exists(targetLink) && Files.isSymbolicLink(targetLink)) {
-        this.context.debug("Deleting symbolic link to be re-created at {}", targetLink);
-        Files.delete(targetLink);
+      if (Files.exists(targetLink)) {
+        if (Files.isSymbolicLink(targetLink)) {
+          this.context.debug("Deleting symbolic link to be re-created at {}", targetLink);
+          Files.delete(targetLink);
+        } else {
+          BasicFileAttributes attr = Files.readAttributes(targetLink, BasicFileAttributes.class,
+              LinkOption.NOFOLLOW_LINKS);
+          if (attr.isOther() && attr.isDirectory() && this.context.getSystemInfo().isWindows()) {
```

Comment:
`isOther()` : 
true if the file something other than a regular file, directory or symbolic link

`isDirectory()` :
true if the file is a directory
(see [Documentation](https://docs.oracle.com/javase/8/docs/api/java/nio/file/attribute/BasicFileAttributes.html))

don't these two cancel each other out and therefore this if would always be false?

Reply:
Ok, looking at the docu it seemed weird, but if it works 👍